aggregator/src/aggregator.rs

```diff
@@ -1387,6 +1387,16 @@ impl VdafOps {
         C: Clock,
         Q: UploadableQueryType,
     {
+        // Shorthand function for generating an Error::ReportRejected with proper parameters.
+        let reject_report = |reason| {
+            Arc::new(Error::ReportRejected(
```

Comment:
Why does this need to be wrapped in an `Arc`?

Reply:
In the context of this code it's to match the existing return type.

For the overall context, it relates to this change https://github.com/divviup/janus/commit/a24d5cbcc075321352b2603dd4b983ab39c2c121, in particular here https://github.com/divviup/janus/blob/main/aggregator/src/aggregator/report_writer.rs#L149. It's to work around the fact that `aggregator::Error` is not `Clone`. We need multiple "copies" of it to fan out transaction errors to a bunch of listening channels.

However I think this can possibly be simplified. I will play around with it and open a PR if we can drop the Arc.

Reply:
Ah, OK. Sometimes I wish that Rust structs were `Clone` by default, but then that's probably a bad idea. Anyway, if we can improve this then I'm all for it, but it's probably not worth too much effort.

Reply:
I was pretty disappointed at having to use an `Arc` here, too -- it added some complexity to the types we work throughout the aggregator implementation.

IIRC, the reason I couldn't just make `aggregator::Error` `Clone` is that some of the included types weren't `Clone`, and they came from other crates so we could not easily make them so. If we want to simplify here, we could try making all included types `Clone`. Alternatively, we could follow the suggestion from the commit Ameer links to & make `aggregator::Error` a wrapper around an `Arc<InnerError>`, where the `InnerError` is equivalent to what `aggregator::Error` is today -- that would let us clone the `aggregator::Error` type without needing the internal types to be `Clone`, but will complicate logic where we discriminate the error type.